python/doc/examples/calibration/bayesian_calibration/plot_gibbs_simus.py

```diff
@@ -0,0 +1,582 @@
+# :math:`\widehat{\vect{\theta}} = \vect{\mu}_n(\widehat\tau).`
+
+def mu_n(tau):
+    x = ot.Point(initial_state)
+    x[p] = tau
+    # trick: posterior conditional mean is computed by the link function
+    return py_link_function_theta(x)[:p]
+
+
+# optimization criterion
+def log_cond_tau_post(tau):
+    x = ot.Point(initial_state)
+    x[p] = tau
+    # replace theta by its conditional posterior mean
+    x[:p] = py_link_function_theta(x)[:p]
+    # compute log conditional posterior of tau
+    ll = py_log_density(x)[0]
+    return ll
+
+
+# %%
+# 1D optimization
+func = lambda X: [-log_cond_tau_post(X[0])]
+problem = ot.OptimizationProblem(ot.PythonFunction(1, 1, func))
+problem.setBounds(ot.Interval([1e-4], [1e4]))
+solver = ot.TNC(problem)
+solver.setStartingPoint([1.0])
+solver.run()
+tauhat = solver.getResult().getOptimalPoint()[0]
+
+# inject result in initialState vector
+x = ot.Point(initial_state)
+x[:p] = mu_n(tauhat)
+x[p] = tauhat
+
+
+# %%
+# Final step : create a :class:`~openturns.MetropolisHastings` object for each block
+
+initialState = x
+
+mi_Y = [i for i in range(p + 1, p + 1 + n)]
+link_function_y = ot.PythonFunction(len(x), 4 * n, py_link_function_y)
+rvmh_Y = ot.RandomVectorMetropolisHastings(RV_Y, initialState, mi_Y, link_function_y)
+
+mi_theta = [i for i in range(p)]
+link_function_theta = ot.PythonFunction(len(x), p + (p + 2) * p, py_link_function_theta)
+rvmh_theta = ot.RandomVectorMetropolisHastings(
+    RV_theta, initialState, mi_theta, link_function_theta
+)
+
+log_pdf_tau = ot.PythonFunction(len(x), 1, py_log_density)
+rwmh_tau = ot.RandomWalkMetropolisHastings(
+    log_pdf_tau, support, initialState, proposal, [p]
+)
+
+# Now, assemble the blocks to create a Gibbs algorithm:
+gibbs = ot.Gibbs([rvmh_Y, rvmh_theta, rwmh_tau])
+
+# %%
+# Launch Algorithm
+sampleSize = 1000
+
+# Joint posterior density sample
+sample = gibbs.getSample(sampleSize)
+
+# compute acceptance rate
+tau_post = np.array(sample[:, p]).ravel()
+acc_rate = (tau_post[1:] != tau_post[:-1]).mean()
+print("Acceptance rate: %s" % acc_rate)
+
+
+# %%
+# Plot posterior distribution marginals
+
+# extract interest parameters
+post_sample = sample.getMarginal([i for i in range(p + 1)])
+post_sample.setDescription(["$\\theta_1$", "$\\theta_2$", "$\\tau$"])
+
+ks = ot.KernelSmoothing()
+posterior = ks.build(post_sample)
+
+# Optional: change the bandwidth
+bandwidth = ks.getBandwidth()
+for k in range(len(bandwidth)):
+    bandwidth[k] = 1.0 * bandwidth[k]
+posterior = ks.build(post_sample, bandwidth)
+
+grid = ot.GridLayout(1, 3)
+grid.setTitle("Bayesian inference")
+xlabs = [r"$\theta_0$", r"$\theta_1$", r"$\tau$"]
+for parameter_index in range(3):
+    graph = posterior.getMarginal(parameter_index).drawPDF()
+    graph.setColors(ot.Drawable.BuildDefaultPalette(2))
+    graph.setLegends(["Posterior"])
+    graph.setXTitle(xlabs[parameter_index])
+    grid.setGraph(0, parameter_index, graph)
+_ = View(grid)
+
```

Comment:
I cannot see what is interesting in this plot. Should'nt the posterior be compared to the prior? What comment may we write so that the reader can see what is the meaning of this graphics?

Reply:
Maybe we can add a vertical line corresponding to the true value of the parameters (as the data are simulated, the true value is known).